src/main/java/io/cdap/plugin/http/common/http/HttpClient.java

```diff
@@ -132,7 +133,14 @@ public CloseableHttpClient createHttpClient(String pageUriStr) throws IOExceptio
       httpClientBuilder.setProxy(proxyHost);
     }
     httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
-
+    ArrayList<Header> clientHeaders = new ArrayList<>();
+    // oAuth2
+    if (config.getOauth2Enabled()) {
+      clientHeaders.add(new BasicHeader("Authorization",
+          "Bearer " + OAuthUtil.getAccessToken(HttpClients.createDefault(), config)
```

Comment:
get access token in a variable first and use String.format. It will be more readable